iceoryx_posh/source/runtime/message_queue_interface.cpp

```diff
@@ -158,9 +158,20 @@ bool MqBase::openMessageQueue(const posix::IpcChannelSide channelSide) noexcept
     m_mq.destroy();
 
     m_channelSide = channelSide;
-    IpcChannelType::create(
-        m_interfaceName, posix::IpcChannelMode::BLOCKING, m_channelSide, m_maxMessageSize, m_maxMessages)
-        .and_then([this](auto& mq) { this->m_mq = std::move(mq); });
+
+    posix::IpcChannelError receivedError{posix::IpcChannelError::UNDEFINED};
+    // In case the server is not present yet (ENOENT), we try again
+    do
+    {
+        IpcChannelType::create(
+            m_interfaceName, posix::IpcChannelMode::BLOCKING, m_channelSide, m_maxMessageSize, m_maxMessages)
+            .and_then([this, &receivedError](auto& mq) {
+                this->m_mq = std::move(mq);
+                receivedError = posix::IpcChannelError::UNDEFINED;
+            })
+            .or_else([&receivedError](posix::IpcChannelError& error) { receivedError = error; });
+
+    } while (receivedError == posix::IpcChannelError::NO_SUCH_CHANNEL);
```

Comment:
If you start more applications than cores are available before roudi is running, the whole system becomes unresponsive. Why not wait a few milliseconds? Why not use the previous behavior?

Reply:
Two valid points. While loops without a sleep burns CPU time and the loop is not needed as the state machine already handles it.